cmd/genallocs/genesis_alloc.go

```diff
@@ -0,0 +1,162 @@
+package genallocs
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"math/big"
+	"os"
+
+	"github.com/dominant-strategies/go-quai/common"
+	"github.com/dominant-strategies/go-quai/params"
+	orderedmap "github.com/wk8/go-ordered-map/v2"
+)
+
+// GenesisAccount is an account in the state of the genesis block.
+type GenesisAccount struct {
+	UnlockSchedule  int                                      `json:"Unlock Schedule"`
+	Address         common.Address                           `json:"Address"`
+	TotalBalance    *big.Int                                 `json:"Amount"`
+	VestedBalance   *big.Int                                 `json:"Vested"`
+	BalanceSchedule *orderedmap.OrderedMap[uint64, *big.Int] `json:"BalanceSchedule"` // Map of blockNumber->balanceUnlocked (at that block).
+}
+
+// Defines the unlockSchedule parameters.
+// Some unlockSchedules have a lumpSum payment, some at TGE, some at 1 year.
+// All unlockSchedules begin regular unlocks after 1 year.
+type unlockSchedule struct {
+	unlockDuration    int    // Total unlocking duration in years. The first year cliff is not part of unlocking.
+	lumpSumPercentage uint64 // One-time percentage unlocked.
+	lumpSumMonth      uint64 // Number of months before lump sum payment.
+	unlockMonthStart  uint64 // Month of first regular unlock.
+}
+
+var unlockSchedules = [4]unlockSchedule{
+	{
+		// schedule0
+		// Immediate unlock: 100% @ Month 0
+		unlockDuration:    0,
+		lumpSumPercentage: 100,
+		lumpSumMonth:      0,
+	},
+
+	{
+		// schedule1
+		// Unlock duration: 5 years
+		// Lump Sum: 2 % @ Month 0
+		// Unlock Start: Month 6
+		unlockDuration:    5,
+		lumpSumPercentage: 2,
+		lumpSumMonth:      0,
+		unlockMonthStart:  6,
+	},
+
+	{
+		// schedule2
+		// Unlock duration: 3 years
+		// Lump Sum: 25% @ Month 0
+		// Unlock Start: Month 13
+		unlockDuration:    3,
+		lumpSumPercentage: 25,
+		lumpSumMonth:      0,
+		unlockMonthStart:  13,
+	},
+
+	{
+		// schedule3
+		// Unlock duration: 3 years
+		// Lump Sum: 25% @ Month 12
+		// Unlock Start: Month 13
+		unlockDuration:    3,
```

Comment:
unlock duration should probably be in months, but what you have appears to work :shrug: